tests/unit/api/main.py

```diff
@@ -1405,9 +1512,113 @@ def port_detach_nic_node_not_free(self):
+                                         'with_request_context')
+
+    def test_list_networks(self):
+        result = json.loads(api.list_networks())
+        for net in result.keys():
+            del result[net]['network_id']
+        assert result == {
+            'manhattan_provider': {'projects':['manhattan']},
+            'manhattan_pxe': {'projects':['manhattan']},
+            'manhattan_runway_provider': {'projects':['manhattan', 'runway']},
+            'manhattan_runway_pxe': {'projects':['manhattan', 'runway']},
+            'pub_default': {'projects':None},
+            'runway_provider': {'projects':['runway']},
+            'runway_pxe': {'projects':['runway']},
+            'stock_ext_pub': {'projects': None},
+            'stock_int_pub': {'projects': None},
+        }
+
+    def test_list_network_attachments(self):
+        api.node_register_nic('runway_node_0', 'r-eth0', 'DE:AD:BE:EF:20:14')
+        api.node_register_nic('manhattan_node_0', 'm-eth0', 'DE:AD:BE:EF:20:14')
+        api.node_connect_network('runway_node_0', 'r-eth0', 'manhattan_runway_pxe')
+        api.node_connect_network('manhattan_node_0', 'm-eth0', 'manhattan_runway_pxe')
+        deferred.apply_networking()
+
+        actual = json.loads(api.list_network_attachments('manhattan_runway_pxe'))
+        expected = {
+            'manhattan_node_0':
+                {
+                    'nic': 'm-eth0',
+                    'channel': get_network_allocator().get_default_channel(),
+                    'project': 'manhattan'
+                },
+            'runway_node_0': 
+                {
+                    'nic': 'r-eth0',
+                    'channel': get_network_allocator().get_default_channel(),
+                    'project': 'runway'
+                }
+            }
+        assert actual == expected
+
+    def test_list_network_attachments_for_project(self):
+        api.node_register('node-99', obm={
+		  "type": "http://schema.massopencloud.org/haas/v0/obm/ipmi",
+		  "host": "ipmihost", 
+		  "user": "root", 
+		  "password": "tapeworm"})
+        api.node_register_nic('node-99', '99-eth0', 'DE:AD:BE:EF:20:14')
+        api.project_create('anvil-nextgen')
+        api.project_connect_node('anvil-nextgen', 'node-99')
+        network_create_simple('hammernet', 'anvil-nextgen')
+        api.node_connect_network('node-99', '99-eth0', 'hammernet')
+        deferred.apply_networking()
+        api.node_register('node-100', obm={
+		  "type": "http://schema.massopencloud.org/haas/v0/obm/ipmi",
+		  "host": "ipmihost", 
+		  "user": "root", 
+		  "password": "tapeworm"})
+        api.node_register_nic('node-100', '100-eth0', 'DE:AD:BE:EF:20:14')
+        api.project_create('anvil-oldtimer')
+        api.project_connect_node('anvil-oldtimer', 'node-100')
+        api.network_grant_project_access('anvil-oldtimer', 'hammernet')
+        api.node_connect_network('node-100', '100-eth0', 'hammernet')
+        deferred.apply_networking()
```

Comment:
This block probably should go in a fixture as well (or use objects in one of the existing fixtures if possible).